integration-tests/src/product.rs

```diff
@@ -22,6 +24,18 @@ impl RegisterProductCommand {
 }
 
 impl RegisterProductCommand {
+    pub(crate) fn json_for_premium(&self, public_key: String) -> Value {
+        let json = self.json();
+
+        match json {
+            Value::Object(mut m) => {
+                m.insert("public_key".to_string(), Value::String(public_key));
+                Value::Object(m)
+            }
+            v => v,
+        }
```

Comment:
Looks a bit cleaner in my opinion.
```suggestion
        let mut json = self.json();
        if let Value::Object(obj) = &mut json {
            obj.insert("public_key".to_string(), Value::String(public_key));
        }
        json
```